akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1309,14 +1309,12 @@ private[akka] trait CallbackWrapper[T] extends AsyncCallback[T] {
     callbackState.get() match {
       case Initialized(cb)          ⇒ cb(arg)
       case list @ NotInitialized(l) ⇒ callbackState.compareAndSet(list, NotInitialized(arg :: l))
-      case Stopped(cb) ⇒
-        lock.unlock()
-        cb(arg)
+      case Stopped(cb)              ⇒ cb(arg)
     }
   }
 
   private[this] def locked(body: ⇒ Unit): Unit = {
     lock.lock()
-    try body finally if (lock.isLocked) lock.unlock()
+    try body finally lock.unlock()
```

Comment:
So, how could this cause the exception in the ticket, didn't that say that there was an `unlock` call to an unlocked lock? So the invoke lock block completed with the lock unlocked inside of it, then the if clause would not match and the lock wouldn't be double-unlocked and cause the exception? What am I missing here?